tests/test_Units.py

```diff
@@ -59,3 +61,50 @@ def test_units(xmlfile, csc, topic):
             "Unit '" + unit.text + "' in " + str(
                 xmlfile.name
             ) + " does not meet astropy standards."
+
+
+@pytest.mark.parametrize("xmlfile,csc,topic", ts_xml.get_xmlfile_csc_topic())
+def test_string_units(xmlfile, csc, topic):
+    """Test that the <Units> field for STRING-type attributes is 'unitless.'
+
+    Parameters
+    ----------
+    xmlfile : `pathlib.Path`
+        Full filepath to the Commands or Events XML file for the CSC.
+    csc : `testutils.subsystems`
+        Name of the CSC
+    topic : `xmlfile.stem`
+        One of ['Commands','Events','Telemetry']
+    """
+    saltype = "SAL" + topic.rstrip("s")
+    # Check for known issues.
+    jira = check_for_issues(csc, topic)
+    if jira:
+        pytest.skip(
+            jira
+            + ": "
+            + str(xmlfile.name)
+            + " <Unit> fields do not conform to astropy standards."
+        )
+    # Test the attribute <Units> fields.
+    with open(str(xmlfile), "r", encoding="utf-8") as f:
+        tree = et.parse(f)
+    root = tree.getroot()
+    for attrib in root.findall(f"./{saltype}/item"):
+        idltype = attrib.find("IDL_Type")
+        if idltype is not None and idltype.text == "string":
+            name = attrib.find("EFDB_Name")
+            unit = attrib.find("Units")
+            # There is a robust skip list,
+            # as many strings represent angles or time.
+            if csc in ["ATPtg", "MTPtg"] and name.text in ts_xml.strings_with_units:
+                assert True
+            else:
+                assert unit.text in ("unitless", "dimensionless"), (
+                    csc
+                    + ": string-type attribute '"
+                    + name.text
+                    + "' has the unit '"
+                    + unit.text
+                    + "'"
```

Comment:
Please consider an f string for readability. Using !r to get the quotes (it means repr):
```
f"{csc}: string-type attribute {name.text!r} has the unit {unit.text!r}"
```

Reply:
Fixed